content/intro-to-storybook/ember/en/data.md

```diff
@@ -71,189 +82,87 @@ export const reducer = (state, action) => {
-  const { tasks } = reducer;
+import { store } from '../store';
+import { pinTask, archiveTask } from '../actions';
 
-  return {
-    tasks: tasks.filter(t => t.state === 'TASK_INBOX' || t.state === 'TASK_PINNED'),
-  };
-};
-
-class TaskList extends Component {
+export default class TaskController extends Controller {
   @action
-  pinTask(taskID) {
-    this.actions.onPinTask(taskID);
+  pinTask(task) {
+    store.dispatch(pinTask(task.id));
   }
 
   @action
-  archiveTask(taskID) {
-    this.actions.onArchiveTask(taskID);
+  archiveTask(task) {
+    store.dispatch(archiveTask(task.id));
   }
 }
-
-export default connect(
-  stateToComputed,
-  dispatch => ({
-    onArchiveTask: id => dispatch(archiveTask(id)),
-    onPinTask: id => dispatch(pinTask(id)),
-  })
-)(TaskList);
 ```
 
-The reason to keep the presentational version of the `TaskList` separate is because it is easier to test and isolate. As it doesn't rely on the presence of a store it is much easier to deal with from a testing perspective. Let's rename `app/components/task-list.stories.js` into `app/components/pure-task-list.stories.js`, and ensure our stories use the presentational version:
-
-```javascript
-// app/components/pure-task-list.stories.js
-import { hbs } from 'ember-cli-htmlbars';
-import { taskData, actionsData } from './task.stories';
-
-export default {
-  title: 'PureTaskList',
-  component: 'task-list',
-  excludeStories: /.*Data$/,
-};
-
-export const defaultTasksData = [
-  { ...taskData, id: '1', title: 'Task 1' },
-  { ...taskData, id: '2', title: 'Task 2' },
-  { ...taskData, id: '3', title: 'Task 3' },
-  { ...taskData, id: '4', title: 'Task 4' },
-  { ...taskData, id: '5', title: 'Task 5' },
-  { ...taskData, id: '6', title: 'Task 6' },
-];
-
-export const withPinnedTasksData = [
-  ...defaultTasksData.slice(0, 5),
-  { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
-];
-
-export const Default = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @pinTask={{fn this.onPinTask}} @archiveTask={{fn this.onArchiveTask}}/></div>`,
-  context: {
-    tasks: defaultTasksData,
-    ...actionsData,
-  },
-});
-
-export const withPinnedTasks = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @pinTask={{fn this.onPinTask}} @archiveTask={{fn this.onArchiveTask}}/></div>`,
-  context: {
-    tasks: withPinnedTasksData,
-    ...actionsData,
-  },
-});
-export const Loading = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @loading={{true}}/></div>`,
-  context: {
-    tasks: [],
-  },
-});
-export const Empty = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}}/></div>`,
-  context: {
-    tasks: [],
-  },
-});
+And we use our earlier created presentational `<TaskList>` component, mount it
+at the route template and connect it with the actions in the controller and the
+data from the route.
```

Comment:
@gossi this could be phrased a bit different. We could probably go with:

" And finally we create a new file called `template.hbs` inside the `tasks` folder, in which we'll add the component we've created in the [previous chapter](/ember/en/composite-component/)". What do you think?